assets/js/SetExpiration.tsx

```diff
@@ -1,155 +1,137 @@
-  alertId: string | null,
-) {
-  if (Array.isArray(expires)) {
-    return;
-  }
-  const expirationConfig = stringify(expires);
-
-  const newConfig = {
-    ...signConfig,
-    expires: expirationConfig,
-    alert_id: alertId,
-  };
-
-  setConfigsFn({
-    [realtimeId]: newConfig,
-  });
-}
-
-function parseDate(str: string | null | undefined) {
-  if (str) {
-    const date = new Date(str);
-
-    if (date.toString() !== 'Invalid Date') {
-      return date;
-    }
-  }
-
-  return null;
-}
-
 interface SetExpirationProps {
   alerts: RouteAlerts;
-  realtimeId: string;
-  signConfig: SignConfig;
-  setConfigs: (x: SignConfigs) => void;
+  expires?: Date | null;
+  alertId?: string | null;
+  onDateChange: (date: Date | null) => void;
+  onAlertChange: (alertId: string | null) => void;
   readOnly: boolean;
   showAlertSelector: boolean;
 }
 
 function SetExpiration({
   alerts,
-  realtimeId,
-  signConfig,
-  setConfigs,
+  expires,
+  alertId,
   readOnly,
+  onDateChange,
+  onAlertChange,
   showAlertSelector,
 }: SetExpirationProps): JSX.Element | null {
-  const isDateTimeSelected = !!signConfig.expires;
-  const isAlertSelected = !!signConfig.alert_id;
+  const isDateTimeSelected = !!expires;
+  const isAlertSelected = !!alertId;
   const isNoAlerts = Object.keys(alerts).length === 0;
-  const headerText = readOnly
-    ? 'Scheduled return to "Auto"'
-    : 'Schedule return to "Auto"';
+  const [selectDate, setSelectDate] = React.useState(isDateTimeSelected);
+  const [selectAlert, setSelectAlert] = React.useState(isAlertSelected);
+  const [returnsToAuto, setReturnsToAuto] = React.useState(
+    isDateTimeSelected || isAlertSelected,
+  );
+
+  const clearExpiration = () => {
+    onDateChange(null);
+    onAlertChange(null);
+  };
+
+  const focusDatePicker = () => {
+    if (selectAlert) {
+      clearExpiration();
+      setSelectAlert(false);
+    }
+
+    setSelectDate(true);
+  };
+
+  const focusAlertPicker = () => {
+    if (selectDate) {
+      clearExpiration();
+      setSelectDate(false);
+    }
+
+    setSelectAlert(true);
+  };
+
+  const toggleReturnToAuto = () => {
+    if (returnsToAuto) {
+      clearExpiration();
+      setReturnsToAuto(false);
+      setSelectDate(false);
+      setSelectAlert(false);
+    } else {
+      setReturnsToAuto(true);
+    }
+  };
 
-  return isDateTimeSelected || isAlertSelected || !readOnly ? (
-    <div>
-      <div>
-        <strong>{headerText}</strong>
-      </div>
-      <form>
-        <label
-          className={isAlertSelected ? 'set_expiration--label-disabled' : ''}
-        >
+  return returnsToAuto || !readOnly ? (
+    <div className="set_expiration">
+      {(!readOnly && (
+        <label>
           <input
             className="mr-1"
-            type="radio"
-            checked={isDateTimeSelected}
-            disabled={isAlertSelected}
+            type="checkbox"
+            checked={returnsToAuto}
+            onClick={() => toggleReturnToAuto()}
```

Comment:
I think here and the `focusAlertPicker` call below can both be listed here as the functions directly (`onClick={toggleReturnToAuto}`), and don't require a new closure to be created.